bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/handler/ZonePlayerHandler.java

```diff
@@ -1468,6 +1477,48 @@ public boolean isOutputLevelFixed() {
         return stateMap.get("MusicSurroundLevel");
     }
 
+    public @Nullable String getCodec() {
+        String codec = stateMap.get("HTAudioIn");
+        switch (codec) {
```

Comment:
The proper approach is to keep the numeric value as state and define state options on the channel to propose clear labels to the user. It would even allow translating in different languages.

Reply:
Correct me if I'm wrong, but state options only apply to UIs correct (I don't use them for anything currently so I'm not very familiar beyond what I've read on the forum)?  Things like rules would still have to parse the numeric values?  Here's my use case and concern.  As part of my testing (and intention for long term use), I have the state of this channel displayed on my Neeo remote.  I like to see that I'm actually decoding in Atmos for those videos (because I have issues sometimes with it not doing what it should).  If I understand this correctly, I would see the numeric value on the Neeo and not the name.  It could also get confusing for people who would use this in a rule.  They should be able to specify "Dolby Atmos" instead of "63".  

Alternative, instead of setting the output to an English output, we could do something like "dolbyAtmos" or "dolbyPCM51" that could be then converted.

Reply:
> Alternative, instead of setting the output to an English output, we could do something like "dolbyAtmos" or "dolbyPCM51" that could be then converted.

Yes, you could keep a conversion to values like NoSignal, DolbyAtmos, ... and then have the state options for better display in UI (and with ability of translations). In documentation, you will have to mention these values (DolbyAtmos, ...).